frontend/GoodGrocer/app/screens/CreateRequest.js

```diff
@@ -0,0 +1,332 @@
+                  type="litres"
+                  onPress={() => setQuantityType("litres")}
+                  selected={currentItem.quantity_type === "litres"}
+                />
+              </View>
+            </View>
+            <TextInput
+              placeholder="Enter preferred brand (empty if none)..."
+              style={{ marginBottom: 5 }}
+              onChange={(text) =>
+                setCurrentItem({ ...currentItem, preferred_brand: text })
+              }
+              value={currentItem.preferred_brand}
+            />
+            <TextInput
+              placeholder="Enter extra notes (empty if none)..."
+              style={{ marginBottom: 5 }}
+              onChange={(text) =>
+                setCurrentItem({ ...currentItem, extra_notes: text })
+              }
+              value={currentItem.extra_notes}
+            />
+            <View style={{ marginTop: 20 }}>
+              {currentItem.image ? (
+                <ImageBackground
+                  style={{
+                    width: Dim.width * 0.25,
+                    height: Dim.width * 0.25,
+                    alignSelf: "center",
+                  }}
+                  source={{
+                    uri: "data:image/png;base64," + currentItem.image,
+                  }}
+                />
+              ) : (
+                <View style={{ alignItems: "center" }}>
+                  <Image
+                    source={require("../assets/grocery-item.png")}
+                    style={{
+                      width: Dim.width * 0.25,
+                      height: Dim.width * 0.25,
+                    }}
+                  />
+                </View>
+              )}
+            </View>
+
+            <TouchableOpacity
+              onPress={async () => await getImagePickerPermissionAsync()}
+              style={{
+                marginVertical: 20,
+                width: Dim.width * 0.5,
+                alignSelf: "center",
+              }}
+            >
+              <Text
+                style={{
+                  color: Colors.darkGreen,
+                  textAlign: "center",
+
+                  fontWeight: "bold",
+                }}
+              >
+                Click here to add an item picture..
+              </Text>
+            </TouchableOpacity>
+            <Button
+              title="Add item"
+              onPress={() => {
+                if (currentItem.name === "") {
+                  Alert.alert("Oops!", "Please enter an item name.");
+                  return;
+                }
+                if (currentItem.quantity === 0) {
+                  Alert.alert("Oops!", "Please enter an item quantity.");
+                  return;
+                }
+                if (currentItem.quantity_type === "") {
+                  Alert.alert("Oops!", "Please enter an item quantity type.");
+                  return;
+                }
+                setItems([...items, currentItem]);
```

Comment:
for currentItem.image, we should add `"data:image/png;base64,"` to the beginning of it before sending or else later when we get the value from the database, it's not going to render correctly

Reply:
or we should change both the ActiveItemCard and ItemCard and add `"data:image/png;base64,"` there